### PR TITLE
Wip/migrate to akka

### PR DESCRIPTION
**Pull request by dotta**
Migrated the codebase to use akka-actor, instead of the old and deprecated scala-actors lib.

Before merging, I'd like to squash all commits but the [last one](https://github.com/dotta/scala-worksheet/commit/0066b0f1becb958bb73b7572f51986ccb245530d) (let me know if you disagree).

### CODE REVIEW

File: org.scalaide.worksheet/src/org/scalaide/worksheet/runtime/WorksheetsManager.scala [hunk 1]
```diff
@@ -62,20 +45,19 @@ private class WorksheetsManager private extends DaemonActor with HasLogger {
...
     val scalaProject = unit.scalaProject
-    worksheetEvaluator.get(scalaProject.underlying.getFullPath) match {
+    worksheets.get(scalaProject.underlying.getFullPath) match {
       case Some(evaluator) => evaluator
       case None =>
-        val evaluator = WorksheetRunner(scalaProject)
-        trapExit = true
-        link(evaluator)
+        val evaluator = context.actorOf(WorksheetRunner.props(scalaProject), "worksheet-runner-for-project-"+scalaProject.underlying.getName)
+        context.watch(evaluator)
```
Comment (dragos) [inline org.scalaide.worksheet/src/org/scalaide/worksheet/runtime/WorksheetsManager.scala]: (as we discussed face to face, this is probably not needed)

**Comment by dragos** [comment]
otherwise, this looks really good! Thanks for taking the time to do this long-overdue migration!

**Comment by dotta** [comment]
@dragos I've updated the code as discussed. Have a look at it, and if you think it's good feel free to :sheep: it

**Comment by dotta** [comment]
ping @dragos

**Comment by dragos** [comment]
Thanks! Godspeed!

**Pull request merged**